website/blog/2019-11-04-binding-react-navigation-5-to-ui-kitten.md

```diff
@@ -0,0 +1,420 @@
+    <Stack.Screen name={AppRoute.AUTH} component={AuthNavigator}/>
+    <Stack.Screen name={AppRoute.HOME} component={HomeNavigator}/>
+  </Stack.Navigator>
+);
+```
+
+<img src="assets/binding-react-navigation-5-to-ui-kitten/bottom-tab-navigator.gif" height="480" />
+
+### Step 4. Drawer menu
+
+At the final stage of this guide, we will describe how to create the drawer navigation. While the top and bottom tabs can be used to present the main product features, a drawer menu can be also used to direct a user to legal information about it, or simply contain quick actions like a logout.
+
+Usually, the drawer menu is available in app on the home screen, so let's add it to `HomeNavigator`. Open [./src/navigation/home.navigator.tsx](https://github.com/artyorsh/react-navigation-ex-demo/blob/complete-exmaples/src/navigation/home.navigator.tsx) file and paste the following code:
+
+```js
+import React from 'react';
+import { createDrawerNavigator } from '@react-navigation/drawer';
+import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
+import { TodoNavigator } from './todo.navigator';
+import { ProfileNavigator } from './profile.navigator';
+import { AppRoute } from './app-routes';
+import { BottomHomeScreen, DrawerHomeScreen, AboutScreen } from '../scenes/home';
+
+const Drawer = createDrawerNavigator();
+const BottomTab = createBottomTabNavigator();
+
+const HomeBottomNavigator = (): React.ReactElement => (
+  <BottomTab.Navigator tabBar={BottomHomeScreen}>
+    <BottomTab.Screen name={AppRoute.TODO} component={TodoNavigator}/>
+    <BottomTab.Screen name={AppRoute.PROFILE} component={ProfileNavigator}/>
+  </BottomTab.Navigator>
+);
+
+export const HomeNavigator = (): React.ReactElement => (
+  <Drawer.Navigator drawerContent={DrawerHomeScreen}>
+    <Drawer.Screen name={AppRoute.HOME} component={HomeBottomNavigator}/>
+    <Drawer.Screen name={AppRoute.ABOUT} component={AboutScreen}/>
+  </Drawer.Navigator>
+);
+```
+
+In this example, we've implemented a Drawer Navigator with `createDrawerNavigator` and used it to display on the Home screen. We have also added `AboutScreen` to demonstrate navigation directly from the Drawer menu.
+
+Just like Top/Bottom tab navigators, the drawer navigator also has a special property for declaring custom drawer view. Use a `drawerContent` property to pass the custom view to the navigator. Open [./src/scenes/home/drawer-home.component.tsx](https://github.com/artyorsh/react-navigation-ex-demo/blob/complete-exmaples/src/scenes/home/drawer-home.component.tsx) file and add the following code:
+
+```js
+import React from 'react';
+import { Drawer, DrawerElement, MenuItemType } from 'react-native-ui-kitten';
+import { AppRoute } from '../../navigation/app-routes';
+import { SafeAreaLayout, SaveAreaInset} from '../../components/safe-area-layout.component';
+import { InfoIcon, LogoutIcon } from '../../assets/icons';
+
+const drawerData: MenuItemType[] = [
```

Comment:
Instead of hardcoding `drawerData`, le's do this:

```js
// Show items for each screen in the navigator
const drawerData = state.routes(route => {
  const options = descriptors[route.key];

  return {
    routeName: route.name,
    title: options.title,
    icon: options.icon,
  }
});

// Add extra menu item for a logout button
drawerData.push({
  title: 'Log Out',
  icon: LogoutIcon,
});
```

Then user can specify `title` and `icon` in screen's `options` prop